libglusterfs/src/inode.c

```diff
@@ -735,7 +735,12 @@ inode_forget_atomic(inode_t *inode, uint64_t nlookup)
         GF_ATOMIC_INIT(inode->nlookup, 0);
     } else {
         inode_lookup = GF_ATOMIC_FETCH_SUB(inode->nlookup, nlookup);
-        GF_ASSERT(inode_lookup >= nlookup);
+        //GF_ASSERT(inode_lookup >= nlookup);
+	if (inode_lookup < nlookup) {
+	    GF_ATOMIC_FETCH_ADD(inode->nlookup, nlookup - inode_lookup);
+	    gf_msg_callingfn(THIS->name, GF_LOG_WARNING, 0, LG_MSG_ASSERTION_FAILED,
```

Comment:
This is a very serious problem. It deserves a critical log level. Basically after this point we know that we may have unsynchronized reference counting with the kernel, which could lead to other unexpected cases. So better give it the relevance it has.

Also, I'm not sure if logging the call stack is relevant in this case, but I'm ok if you prefer that.

```suggestion
	    gf_msg_callingfn(THIS->name, GF_LOG_CRITICAL, 0, LG_MSG_ASSERTION_FAILED,
```